nextflow_aws_batch_trigger.py

```diff
@@ -0,0 +1,314 @@
+import boto3
+import base64
+import argparse
+import datetime
+import time
+import logging
+
+now = datetime.datetime.now()
+
+logging.getLogger().setLevel(logging.DEBUG)
+logging.basicConfig(filename="batch_trigger.log", level=logging.INFO)
+
+
+def create_launch_template(
+    ec2_client, launch_template_name, key_name, s3fs_mount, s3_bucket
+):
+    # Use the describe_launch_templates() method to retrieve a list of all launch templates
+    launch_template_response = ec2_client.describe_launch_templates()
+
+    # Extract the LaunchTemplates list from the response
+    launch_templates = launch_template_response["LaunchTemplates"]
+
+    launch_template_names = [i["LaunchTemplateName"] for i in launch_templates]
+
+    if launch_template_name not in launch_template_names:
+        user_data = (
+            "MIME-Version: 1.0\nContent-Type: multipart/mixed; boundary="
+            "==MYBOUNDARY=="
+            "\n\n --==MYBOUNDARY"
+            "==\nContent-Type: text/x-shellscript; charset="
+            "us-ascii"
+            "\n\n#!/bin/bash\n#!/bin/bash -xe\nsudo "
+            "amazon-linux-extras install epel -y\nsudo yum install s3fs-fuse -y\nmkdir "
+            + s3fs_mount
+            + "\nchmod 777 "
+            + s3fs_mount
+            + "\nsudo s3fs "
+            + s3_bucket
+            + " "
+            + s3fs_mount
+            + " -o allow_other -o "
+            "umask=000 "
+            "-o iam_role=auto"
+            "\n\n--==MYBOUNDARY==-- "
+        )
+
+        encoded_user_data = base64.b64encode(user_data.encode()).decode()
+
+        response = ec2_client.create_launch_template(
+            LaunchTemplateName=launch_template_name,
+            VersionDescription="Launch template for running Nextflow on AWS Batch",
+            LaunchTemplateData={"KeyName": key_name, "UserData": encoded_user_data},
+        )
+        logging.info(
+            f"{now.strftime('%Y-%m-%d %H:%M:%S')} The Launch Template: {launch_template_name} has been created."
```

Comment:
As mentioned above, you shoudlnt need to use `now` object since logging will include it. Same for logging statements bellow this.

Reply:
Removed, as suggested.